common/src/primitives/id.rs

```diff
@@ -133,6 +133,57 @@ pub trait Idable {
     fn get_id(&self) -> Id<Self::Tag>;
 }
 
+impl<T: Idable> Idable for &T {
+    type Tag = T::Tag;
+    fn get_id(&self) -> Id<Self::Tag> {
+        (*self).get_id()
+    }
+}
+
+/// An object together with its pre-calculated ID.
+///
+/// This only allows immutable access to the underlying object to prevent it from going out of sync
+/// with the ID, which is calculated for its contents. Getting an ID is nearly cost-free.
```

Comment:
I think that the "near" word here is somewhat misleading. I have to immediately check if there is still some computation present. 🙃